app/src/main/java/com/example/filemanager/ListAdapter.kt

```diff
@@ -9,18 +9,19 @@ import android.widget.TextView
 import androidx.core.content.ContextCompat
 import androidx.recyclerview.widget.RecyclerView
 
-internal class ListAdapter(
+class ListAdapter(
     private val context: Context,
```

Comment:
I think you dont need to pass the context as argument, you can access it from view, please check line 27 `LayoutInflater.from(parent.context)` you are getting the context from parent `ViewGroup`, you can use that one and remove this one.